### DISCUSSION TITLE
OS Update without new Install?

### DISCUSSION THREAD

**Discussion by Parateam** [freakent/dbus-mqtt-devices#95] [Q&A]
Hello, is it possible to get an permanent installation after an update of the Victron OS?

I now that only the /DATA partition is not touched during that.

Peter from Munich

**Comment by freakent** [comment]
Hi Peter, unfortunately the latest VenusOS updates remove the libraries that the driver relies on. At the moment the best I can suggest is just to run the setup.sh script again. No need to reinstall everything.

Eventually I am am going to remove the Yaml dependency altogther, so that will address this issue.

**Comment by Parateam** [comment]
Is it a good advice to put the install.sh in the rc.local? This would make an install at every reboot?

The driver dbus-opendtu is making it in that way

**Comment by freakent** [comment]
I did consider that but I didn't feel it's a great long term solution. If it works for you in the short term then go ahead. Let us know how it goes for you. I'll set my test system up the same and see what happens.

**Reply by Parateam** [reply]
OK thanks. Also for yout work!

**Comment by freakent** [comment]
I am thinking to have the just the dependencies in a seperate script setup-dependencies.sh and call that from rc.local instead.

```
#!/bin/sh
BASE=$(dirname $(dirname $(realpath "$0")))

echo "dbus-mqtt-devices: Checking to see if Python's Pip is installed"
python -m pip --version
piperr=$?
if [ "$piperr" -ne 0 ]; then
    opkg update && opkg install python3-pip
fi

echo "dbus-mqtt-devices: Checking to see if python3-misc library is installed"
if [ -z "`opkg list-installed | grep python3-misc`" ]; then
    opkg update && opkg install python3-misc
fi

echo "dbus-mqtt-devices: Pip install module dependencies"
python -m pip install -r $BASE/requirements.txt
```

**Reply by Parateam** [reply]
OK, i have a look at that way and try it.
Thanks

**Comment by Parateam** [comment]
Hello, i have the 0.6.5 installed and it still runs after an update of the venus OS.

What are the changes in your beta? No more yaml and only system libraries? So no components must installed? Is it working with pvinverter service?

Thanks, Peter

**Reply by freakent** [reply]
Thanks, that's good feedback. 

I removed the beta release as it was not functioning properly. The next release will move to a different format for defining services. Currently it's one big YAML file, but this will become individual ini files. Removing the YAML dependency is just one of the advantages.
